Remove dependencies on packages we build here

rynowak opened the pull request:
This change restores automated dependency from into this repo.

dotnet/aspnetcore#19033 migrated some dependencies into this repo but
did not update the version files. This caused generation to dependency
PRs to fail.

This change fixes the dependency tracking and manually updates
dependencies.

@JunTaoLuo

rynowak committed:
Remove dependencies on packages we build here

This change restores automated dependency from into this repo.

dotnet/aspnetcore#19003 migrated some dependencies into this repo but
did not update the version files. This caused generation to dependency
PRs to fail.

This change fixes the dependency tracking and manually updates
dependencies.

rynowak commented on eng/Version.Details.xml:
```diff
@@ -57,352 +57,316 @@
-      <Sha>bc91b865ad4e746e7e3f281351e8d4a8d1dc80a0</Sha>
+      <Sha>8fb38ab52b8f873109b6e87b4f04226d6a892ee4</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.Extensions.Options.DataAnnotations" Version="5.0.0-preview.2.20121.3" CoherentParentDependency="Microsoft.AspNetCore.Razor.Language">
```
The diff is bad here because XML. The list of things I removed is based on the package list in 19033.

If there were still 'extra' references then the dependency update would fail, so I know I got them all.

JunTaoLuo commented:
Strange. I made this change in #19033 in the commit https://github.com/dotnet/aspnetcore/pull/19033/commits/7d1f5d8c0b290043180d345f688049a17c23eddf. Maybe it got lost in the merge commits later on.

JunTaoLuo reviewed (changes requested):
We should also remove all the .Sources packages. They are now in the /src/Shared. They still exist in extensions but will be removed when types are migrated to dotnet/runtime.

rynowak commented:
> We should also remove all the .Sources packages. They are now in the /src/Shared. They still exist in extensions but will be removed when types are migrated to dotnet/runtime.

So if I understand you correctly, these are still published from extensions, but we don't actually depend on them?

JunTaoLuo commented:
Yes

rynowak committed:
Remove sources packages from dependency list

This code has already migrated to this repo. The extensions repo still
publishes these packages, but they are not consumed here. Removing the
references from dependency lists will avoid a future break when they go
away from extensions.

rynowak commented:
@JunTaoLuo updated
